packages/react-native-gesture-handler/src/v3/hooks/events/useGestureHandlerEvent.ts

```diff
@@ -0,0 +1,54 @@
+import { CALLBACK_TYPE } from '../../../handlers/gestures/gesture';
+import {
+  GestureStateChangeEvent,
+  GestureTouchEvent,
+  GestureUpdateEvent,
+} from '../../../handlers/gestureHandlerCommon';
+import { isStateChangeEvent, isTouchEvent, runWorklet } from '../utils';
+import { Reanimated } from '../../../handlers/gestures/reanimatedWrapper';
+
+export function useGestureHandlerEvent(
+  config: any,
+  shouldUseReanimated: boolean
+) {
+  const onGestureHandlerEvent = (
+    event: GestureUpdateEvent | GestureStateChangeEvent | GestureTouchEvent
+  ) => {
+    'worklet';
+    if (isStateChangeEvent(event) || isTouchEvent(event)) {
+      return;
+    }
+
+    runWorklet(CALLBACK_TYPE.UPDATE, config, event);
+
+    if (config.onChange && config.changeEventCalculator) {
+      runWorklet(
+        CALLBACK_TYPE.CHANGE,
+        config,
+        config.changeEventCalculator?.(
+          event
+          //   lastUpdateEvent.value[config.handlerTag]
+        )
+      );
+
+      //   lastUpdateEvent.value[gesture.handlerTag] = event;
+    }
+  };
+
+  if (shouldUseReanimated) {
+    const handlers = {
+      onChange: config.onChange,
+      onUpdate: config.onUpdate,
+    };
+
+    const { doDependenciesDiffer } = Reanimated!.useHandler(handlers);
+
+    return Reanimated!.useEvent(
```

Comment:
We may want to investigate whether there's a difference (performance-wise) of calling `useEvent` separately for every event vs. calling it once and handling all events there.

Reply:
I discussed it with Reanimated team and they said that it shouldn't be a big deal. I think we can leave it as it is for now and later we can check whether it is a big performance issue or not 😅